packages/cli/src/ui/components/InputPrompt.tsx

```diff
@@ -584,56 +584,82 @@ export const InputPrompt: React.FC<InputPromptProps> = ({
     const usedWidth = stringWidth(textBeforeCursor);
     const remainingWidth = Math.max(0, inputWidth - usedWidth);
 
+    const ghostTextLinesRaw = ghostSuffix.split('\n');
+    const firstLineRaw = ghostTextLinesRaw.shift() || '';
+
     let inlineGhost = '';
-    const ghostCodePoints = toCodePoints(ghostSuffix);
-    let ghostUsedWidth = 0;
-    let inlineEndIndex = 0;
-
-    for (let i = 0; i < ghostCodePoints.length; i++) {
-      const char = ghostCodePoints[i];
-      if (char === '\n') {
-        break;
-      }
+    let remainingFirstLine = '';
 
-      const charWidth = stringWidth(char);
-      if (ghostUsedWidth + charWidth > remainingWidth) {
-        break;
+    if (stringWidth(firstLineRaw) <= remainingWidth) {
```

Comment:
the previous wrapping made it a bit hard to read when the completion wrapped. this is generally boilerplate word wrapping. we could refactor this in a followup to better leverage word wrap support already in text-buffer.ts